commerce-price-list-service/src/main/java/com/liferay/commerce/price/list/service/impl/CommercePriceEntryServiceImpl.java

```diff
@@ -296,4 +314,51 @@ public CommercePriceEntry upsertCommercePriceEntry(
 			skuExternalReferenceCode, serviceContext);
 	}
 
+	private void _checkCommerceCatalogPermission(long groupId, String actionId)
+		throws PortalException {
+
+		CommerceCatalog commerceCatalog =
+			_commerceCatalogService.fetchCommerceCatalogByGroupId(groupId);
+
+		if (commerceCatalog == null) {
+			throw new PrincipalException();
+		}
+
+		_commerceCatalogModelResourcePermission.check(
+			getPermissionChecker(), commerceCatalog, actionId);
+	}
+
+	private void _checkCommerceCatalogPermissionByCommercePriceListId(
+			long commercePriceListId, String actionId)
+		throws PortalException {
+
+		CommercePriceList commercePriceList =
+			commercePriceListLocalService.fetchCommercePriceList(
+				commercePriceListId);
+
+		if (commercePriceList == null) {
+			throw new NoSuchPriceListException();
+		}
+
+		CommerceCatalog commerceCatalog =
```

Comment:
seems like we might as well call `_checkCommerceCatalogPermission(commercePriceList.getGroupId(), actionId)` at this point.